INP001 check should respect `ruff.src` setting or have it configured separately

A lot of projects nest code under `src` directory by adding `src` to PYTHONPATH:
```
src/
  some_package/
    __init__.py
    ...
  main.py

pyproject.toml
```
```bash
ruff .
```
```bash
src\main.py:1:1: INP001 File `\src\main.py` is part of an implicit namespace package. Add an `__init__.py`.
Found 1 error(s)
```
Ruff version: 0.0.225
Ruff configuration:
```toml
[tool.ruff]
select = ["INP"]
src = ["src"]
```

A solution to this might be using `ruff.src` or different settings to exclude files contained directly in that directory  
But this might be an issue for packages that would want to use `ruff.src` for example for `isort` checks but would still want their package directory to be checked by `INP001`

Adding `__init__.py` actually breaks some tools like mypy:
```py
# src/some_package/test.py
def a() -> int:
    return 42
```
```
# main.py
from some_package.test import a

reveal_type(a())
```

Without `__init__.py`:
```
$ mypy .
src\main.py:3: note: Revealed type is "builtins.int"
Success: no issues found in 3 source files
```

With `__init__.py`:
```
$ mypy .
src\main.py:1: error: Skipping analyzing "some_package.test": module is installed, but missing library stubs or py.typed marker  [import]
src\main.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src\main.py:3: note: Revealed type is "Any"
Found 1 error in 1 file (checked 4 source files)
```

--- a/src/checkers/filesystem.rs
+++ b/src/checkers/filesystem.rs
@@ -13,7 +13,7 @@ pub fn check_file_path(
 
     // flake8-no-pep420
     if settings.rules.enabled(&Rule::ImplicitNamespacePackage) {
-        if let Some(diagnostic) = implicit_namespace_package(path, package) {
+        if let Some(diagnostic) = implicit_namespace_package(path, package, &settings.src) {
             diagnostics.push(diagnostic);
         }
     }

--- a/src/rules/flake8_no_pep420/rules.rs
+++ b/src/rules/flake8_no_pep420/rules.rs
@@ -1,4 +1,4 @@
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 use ruff_macros::derive_message_formats;
 
@@ -19,8 +19,19 @@ impl Violation for ImplicitNamespacePackage {
 }
 
 /// INP001
-pub fn implicit_namespace_package(path: &Path, package: Option<&Path>) -> Option<Diagnostic> {
-    if package.is_none() && path.extension().map_or(true, |ext| ext != "pyi") {
+pub fn implicit_namespace_package(
+    path: &Path,
+    package: Option<&Path>,
+    src: &[PathBuf],
+) -> Option<Diagnostic> {
+    if package.is_none()
+        // Ignore `.pyi` files, which don't require an `__init__.py`.
+        && path.extension().map_or(true, |ext| ext != "pyi")
+        // Ignore any files that are direct children of a source directory (e.g., `src/manage.py`).
+        && !path
+            .parent()
+            .map_or(false, |parent| src.iter().any(|src| src == parent))
+    {
         #[cfg(all(test, windows))]
         let path = path
             .to_string_lossy()

